packages/lodestar/src/tasks/tasks/checkPeerAliveTask.ts

```diff
@@ -0,0 +1,66 @@
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {INetwork} from "../../network";
+import {IReputationStore} from "../../sync/IReputation";
+import {ILogger} from "@chainsafe/lodestar-utils";
+import {handlePeerMetadataSequence} from "../../sync/utils/reputation";
+
+export interface ICheckPeerAliveModules {
+  network: INetwork;
+  reps: IReputationStore;
+  logger: ILogger;
+}
+
+/**
+ * Periodically ping peers.
+ *     If cannot ping them then disconnect.
+ *     If its sequence is ahead of our store, issue metadata request to update us.
+ */
+export class CheckPeerAliveTask {
+  private readonly config: IBeaconConfig;
+  private readonly network: INetwork;
+  private readonly logger: ILogger;
+
+  private reps: IReputationStore;
+  private interval?: NodeJS.Timeout;
+
+  public constructor(config: IBeaconConfig, modules: ICheckPeerAliveModules) {
+    this.config = config;
+    this.network = modules.network;
+    this.logger = modules.logger;
+    this.reps = modules.reps;
+  }
+
+  public async start(): Promise<void> {
+    this.interval = setInterval(
+      this.run,
+      1 * this.config.params.SLOTS_PER_EPOCH * this.config.params.SECONDS_PER_SLOT * 1000
```

Comment:
I don't like sending batch of ping requests, maybe store last sent ping timestamp and send another if sufficient time has passed. That said, we can leave for now, I'm migrating metadata to peerstore anyways